sdk/test/AzureTest/Scrubbing.swift

```diff
@@ -0,0 +1,120 @@
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+import Foundation
+import DVR
+
+public extension Filter {
+    static let subscriptionIDRegex = try! NSRegularExpression(pattern: "(/(subscriptions))/[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}", options: [.caseInsensitive])
+    static let graphSubscriptionIDRegex = try! NSRegularExpression(pattern: "(https://(graph.windows.net))/[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}", options: [.caseInsensitive])
+    
+    static let standard: Filter = {
+        var standardFilter = Filter()
+        let headers = ["Authorization" : FilterBehavior.remove]
+        standardFilter.beforeRecordRequest = scrubSubscriptionIDs
+        standardFilter.beforeRecordResponse = scrubSubscriptionIDs
+        return standardFilter
+    }()
+    
+    static func scrubSubscriptionId(from string: String?, embeddedStringKeys: [String] = ["tenantID","objectId"]) -> String? {
+        
+        
+        
+        let embeddedIDRegex = { (_ key : String) in
+            try! NSRegularExpression(pattern: "\"\(key)\":\"/[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}\"", options: [.caseInsensitive])
+        }
+        
+        guard var editedString = string else {
+            return nil
+        }
+        var range = NSRange(location: 0, length: editedString.utf8.count)
+        if Filter.subscriptionIDRegex.numberOfMatches(in: editedString, options: [], range: range) > 0 {
+            editedString = Filter.subscriptionIDRegex.stringByReplacingMatches(in: editedString, options: [], range: range, withTemplate: "$1/99999999-9999-9999-9999-999999999999")
+        }
+        range = NSRange(location: 0, length: editedString.utf8.count)
+        if Filter.graphSubscriptionIDRegex.numberOfMatches(in: editedString, options: [], range: range) > 0 {
+            editedString = Filter.graphSubscriptionIDRegex.stringByReplacingMatches(in: editedString, options: [], range: range, withTemplate: "$1/99999999-9999-9999-9999-999999999999")
+        }
+        
+        for key in embeddedStringKeys {
+            let embeddedRegex = embeddedIDRegex(key)
+            range = NSRange(location: 0, length: editedString.utf8.count)
+            if embeddedRegex.numberOfMatches(in: editedString, options: [], range: range) > 0 {
+                editedString = embeddedRegex.stringByReplacingMatches(in: editedString, options: [], range: range, withTemplate: "\"\(key)\":\"/99999999-9999-9999-9999-999999999999\"")
+            }
+        }
+        return editedString
+        
+        
+    }
+    
+    static func scrubSubscriptionIDs(request: URLRequest) -> URLRequest {
```

Comment:
Per swift naming guidelines, this should probably be something like `scrubSubscriptionsIds(for request: URLRequest)` or `scrubSubscriptionsIds(forRequest request: URLRequest)`